src/resmom/linux/mom_mach.c

```diff
@@ -5548,16 +5547,18 @@ mom_get_sample(void)
 	unsigned long long 	starttime;
 	int			nskipped = 0;
 	extern time_t		time_last_sample;
-	char			*stat_str;
+	char			*stat_str = NULL;
 
 	DBPRT(("%s: entered\n", __func__))
 	if (pdir == NULL)
 		return PBSE_INTERNAL;
 
 #if MOM_CPUSET
 	if (((pidcache = pidcache_getarena()) == NULL) && pidcache_needed()) {
-		if ((pidcache = pidcache_create()) == NULL)
+		if ((pidcache = pidcache_create()) == NULL) {
 			log_err(errno, __func__, "PID cache create");
+			return PBSE_SYSTEM;
+		}
```

Comment:
I added this check to validate calloc failure that could happen within pidcache_create(). Now revisiting the code I see we already safeguarded all our pidcache usage code with a (pidcache != NULL) check. So by removing this return even on calloc failure we will continue with considering all the processes running the system to be monitored by MoM as we used to do earlier.